pkg/controller/job/job_controller.go

```diff
@@ -772,6 +772,21 @@ func (jm *Controller) syncJob(ctx context.Context, key string) (rErr error) {
 	}
 
 	activePods := controller.FilterActivePods(pods)
+
+	// As part of KEP-3939,
+	// we will allow for opt-in control of including terminating pods
+	// in our activePod list.
+	// This works because we pass this list into managePods which counts the list of activePods.
+	var terminatingPods *int32 = pointer.Int32(0)
+	if feature.DefaultFeatureGate.Enabled(features.JobPodReplacementPolicy) {
+		terminatingPods = controller.CountTerminatingPods(pods)
+	}
+	if recreatePodsWhenFailed(&job) {
```

Comment:
can this be inside the feature gate conditional?

Reply:
updated.